GooglePlayInstant/Editor/QuickDeploy/AssetBundleVerifierWindow.cs

```diff
@@ -61,41 +77,69 @@ private void StartAssetBundleVerificationDownload()
 #endif
         }
 
-        private void GetAssetBundleInfoFromDownload()
+        internal void HandleAssetBundleVerifyState(AssetBundleVerifyState state, UnityWebRequest webRequest)
         {
-            var bundle = DownloadHandlerAssetBundle.GetContent(_webRequest);
+            switch (state)
+            {
+                case AssetBundleVerifyState.DestinationError:
+                    AssetBundleDownloadIsSuccessful = false;
+                    ErrorDescription = "Cannot connect to destination host. See Console log for details.";
+                    // No need to log since debugging information in this case is logged from thrown exception.
+                    break;
+                case AssetBundleVerifyState.WebRequestError:
+                    AssetBundleDownloadIsSuccessful = false;
+                    ErrorDescription = webRequest.error;
+                    Debug.LogErrorFormat(WebRequestErrorFormatMessage, AssetBundleUrl,
+                        ErrorDescription);
+                    break;
+                case AssetBundleVerifyState.BundleError:
+                    AssetBundleDownloadIsSuccessful = false;
+                    ErrorDescription = "Error extracting AssetBundle. See Console log for details.";
+                    // No need to log since debugging information in this case is automatically logged by Unity.
+                    break;
+                case AssetBundleVerifyState.DownloadSuccess:
+                    AssetBundleDownloadIsSuccessful = true;
+                    _numOfMegabytes = ConvertBytesToMegabytes(webRequest.downloadedBytes);
+                    var scenes = _bundle.GetAllScenePaths();
+                    _mainScene = (scenes.Length == 0) ? "No scenes in AssetBundle" : scenes[0];
+                    break;
+            }
+        }
+
+        private AssetBundleVerifyState GetAssetBundleVerifyStateInfoFromDownload()
+        {
+            if (!_webRequest.isDone)
+            {
+                return AssetBundleVerifyState.InProgress;
+            }
+
+            try
+            {
+                _bundle = DownloadHandlerAssetBundle.GetContent(_webRequest);
+            }
+            catch (InvalidOperationException e)
+            {
+                Debug.LogError(e.ToString());
```

Comment:
Something like this maybe
`Debug.LogErrorFormat("Failed to obtain AssetBundle content: {0}", e);`

Reply:
Resolved.